### PR TITLE
[clang][NFC] Trim license header comments to 81 characters

### PR DESCRIPTION
**Pull request by steakhal**
clang-format would format these headers poorly by splitting it into multiple lines.

```regex
//=.{78,}
```

### CODE REVIEW

**Review by NagyDonat — APPROVED** [review]

File: clang/unittests/ASTMatchers/Dynamic/VariantValueTest.cpp [hunk 1]
```diff
@@ -1,10 +1,10 @@
-//===- unittest/ASTMatchers/Dynamic/VariantValueTest.cpp - VariantValue unit tests -===//
+// unittest/ASTMatchers/Dynamic/VariantValueTest.cpp - VariantValue unit tests
```
Comment (pogo59) [inline clang/unittests/ASTMatchers/Dynamic/VariantValueTest.cpp:1]: Repeating the full path name seems unnecessary? (In various files.)

**Comment by dwblaikie** [comment]
+1 to @pogo59's comment about pruning complete paths - I suspect they're in the minority. Might be worth checking whether the `===` at the start and end is markup for any particular thing (I /think/ the `-*- C++ -*-` is load bearing for some editors to inform them this `.h` file is C++ not C, so I'm not sure about some other features of those top-of-file comments).

**Comment by steakhal** [comment]
> +1 to @pogo59's comment about pruning complete paths - I suspect they're in the minority. Might be worth checking whether the `===` at the start and end is markup for any particular thing (I /think/ the `-*- C++ -*-` is load bearing for some editors to inform them this `.h` file is C++ not C, so I'm not sure about some other features of those top-of-file comments).

According to the [LLVM Coding Standard](https://llvm.org/docs/CodingStandards.html#file-headers), it should look like this:
```
//===-- llvm/Instruction.h - Instruction class definition -------*- C++ -*-===//
//
// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
//
//===----------------------------------------------------------------------===//
///
/// \file
/// This file contains the declaration of the Instruction class, which is the
/// base class for all of the VM instructions.
///
//===----------------------------------------------------------------------===//
```
And also explicitly notes:
> This tag ["-*- C++ -*-"] is not necessary in .cpp files. The name of the file is also on the first line, along with a very short description of the purpose of the file.

I could probably interpret this as we could drop the path to the file, but keep the last part.

How should I proceed?

**Comment by pogo59** [comment]
I grepped `clang` for "^/+=[^/]*/.*\.(h|cpp) " and found only 35 files with directory components. So, the majority of files have only the filename in the header line. The example in the Coding Standard does not reflect common practice.

I think it's fine to remove everything but the filename.

File: clang/unittests/ASTMatchers/Dynamic/VariantValueTest.cpp (hunk 1 above)
Comment (steakhal) [reply clang/unittests/ASTMatchers/Dynamic/VariantValueTest.cpp]: Fixed.

**Comment by steakhal** [comment]
> +1 to @pogo59's comment about pruning complete paths - I suspect they're in the minority. Might be worth checking whether the `===` at the start and end is markup for any particular thing (I /think/ the `-*- C++ -*-` is load bearing for some editors to inform them this `.h` file is C++ not C, so I'm not sure about some other features of those top-of-file comments).

I'm using now the filenames for the files I touch in this patch.
I made sue that the `-*- C++ -*-` is present for headers, and I kept it for C++ files if there was room for it and was present in the past.

**Review by pogo59 — APPROVED** [review]
LGTM

**Pull request merged**